src/datasets/arrow_dataset.py

```diff
@@ -1577,6 +1577,7 @@ def map(
         disable_nullable: bool = False,
         fn_kwargs: Optional[dict] = None,
         num_proc: Optional[int] = None,
+        sequential: Optional[bool] = None,
```

Comment:
Though the context of the PR was just to load sequentially, I couldn't find a good way to check that all shards were already cached. Instead I expose this value to map. Is there a way to check before the transformation that the shard are all already cached?